test-gui/src/package/view-decoded-linear-position-plot/DecodedLinearPositionPlotView.tsx

```diff
@@ -102,19 +101,28 @@ const DecodedLinearPositionPlotView: FunctionComponent<DecodedLinearPositionProp
         context.imageSmoothingEnabled = false
         context.drawImage(canvas, displayRange[0], 0, displayRange[1] - displayRange[0], canvas.height, 0, 0, panelWidth, panelHeight)
         if (showObservedPositionsOverlay && scaledObservedPositions !== undefined) {
-            const visibleObserved = scaledObservedPositions.slice(downsampledStart, downsampledEnd + 1)
+            const verticalEpsilon = 0.005
+            const alignedStart = downsampledStart * scaleFactor
+            const alignedEnd = downsampledEnd * scaleFactor
+            const visibleObserved = scaledObservedPositions.slice(alignedStart, alignedEnd + 1)
             const xStepSize = visibleObserved.length > 0 ? panelWidth / (visibleObserved.length) : 1
-            let lastP = visibleObserved[0]
             context.strokeStyle = (observedPositionsStyle ?? '#000000')
             context.lineWidth = 2
+            let lastV = visibleObserved[0]  // TODO: would it be better to set the threshold as something like 2-3 pixels?
+            let lastX = -5   // avoids performance loss from drawing sub-pixel points on top of each other. Negative value to handle first point right
             context.beginPath()
             visibleObserved.forEach((v, i) => {
-                Math.abs(v - lastP) > .005 ? context.moveTo(i * xStepSize, panelHeight * v) : context.lineTo(i * xStepSize, panelHeight * v)
-                lastP = v
+                const x = i * xStepSize
+                const deltaV = Math.abs(v - lastV)
+                if ((Math.floor(lastX) !== Math.floor(x)) || (deltaV > verticalEpsilon)) {
```

Comment:
Logically equivalent to saying "Skip any observation which maps to the same X position as the previous one and less than 0.5% difference in y-position", which in effect means we'll only actually draw one point per pixel (which is much gentler on the drawing routines).

Note that we want to include points that map to the same x coordinate if they cross a discontinuity--this makes sure the animal is represented on both sides of the boundary when it crosses one within the same sample.